### PR TITLE
kqueue() Fix memory leak

### PR DESCRIPTION
**Pull request by jpereira**

### CODE REVIEW

File: src/common/kqueue.c [hunk 1]
```diff
@@ -144,13 +144,14 @@ kqueue(void)
 #endif
 
     kq = calloc(1, sizeof(*kq));
-    if (kq == NULL)
-        return (-1);
+    if (kq == NULL) return (-1);
```
Comment (arr2036) [inline src/common/kqueue.c:147]: Please don't reformat stuff to FR style.

Comment (jpereira) [reply src/common/kqueue.c]: done

File: src/common/kqueue.c [hunk 2]
```diff
@@ -150,7 +150,9 @@ kqueue(void)
     tracing_mutex_init(&kq->kq_mtx, NULL);
 
     if (kqops.kqueue_init(kq) < 0) {
+    error:
         free(kq);
+        tracing_mutex_destroy(&kq->kq_mtx);
```
Comment (arr2036) [inline src/common/kqueue.c:155]: Destroying the mutex _AFTER_ freeing the kq?

Comment (jpereira) [reply src/common/kqueue.c:154]: :exploding_head: It was a _typo_ :laughing: . fixed

**Pull request merged**